Assets/Scripts/Board/Graph.cs

```diff
@@ -0,0 +1,80 @@
+using UnityEngine;
+using System.Collections.Generic;
+
+public class Graph
+{
+    public enum Building
+    {
+        Settlement,
+        City,
+    }
+
+    public class Node { }
+
+    public class VertexNode : Node
+    {
+        public List<EdgeNode> edges = new();
+        public Building building;
+        public string owner;
+
+        public VertexNode() { }
+
+        public void AddEdge(EdgeNode edge)
+        {
+            edges.Add(edge);
+        }
+    }
+
+    public class EdgeNode : Node
+    {
+        public List<VertexNode> vertices = new();
+        public bool hasRoad = false;
```

Comment:
Using a bool switch for a one-time setter isn't great IMO. It makes sense for the TileNode somewhat in that it changes frequently, but roads will only ever change once, i.e. once the EdgeNode has an owner, that owner will never change, so we can just use the owner to determine if the edge is occupied by a road